release_automation.sh

```diff
@@ -336,7 +336,18 @@ execute "git" "switch" "-c" "gutenberg/integrate_release_$VERSION_NUMBER"
 ohai "Update GitHub organization and gutenberg-mobile ref"
 test -f "Podfile" || abort "Error: Could not find Podfile"
 sed -i'.orig' -E "s/wordpress-mobile(\/gutenberg-mobile)/$MOBILE_REPO\1/" Podfile || abort "Error: Failed updating GitHub organization in Podfile"
-sed -i'.orig' -E "s/gutenberg :(commit|tag) => '(.*)'/gutenberg :commit => '$GB_MOBILE_PR_REF'/" Podfile || abort "Error: Failed updating gutenberg ref in Podfile"
+# The Gutenberg reference can be in two different formats in Podfile, hence we have to detect which one is being used before updating it.
+#   - Old format: gutenberg :tag => 'v1.0.0'
+#   - New format: gutenberg tag: 'v1.0.0'
+if ( grep -qE "gutenberg :(commit|tag) => '(.*)'" Podfile ) then
+    # Old format (WordPress-iOS 20.0 and older)
+    sed -i'.orig' -E "s/gutenberg :(commit|tag) => '(.*)'/gutenberg :commit => '$GB_MOBILE_PR_REF'/" Podfile || abort "Error: Failed updating gutenberg ref in Podfile (old format)"
+elif ( grep -qE "gutenberg (commit|tag): '(.*)'" Podfile ) then
+    # New format (WordPress-iOS 20.1 and newer)
+    sed -i'.orig' -E "s/gutenberg (commit|tag): '(.*)'/gutenberg :commit => '$GB_MOBILE_PR_REF'/" Podfile || abort "Error: Failed updating gutenberg ref in Podfile (new format)"
```

Comment:
I noticed that we might rather use the new format here, i.e.:
`gutenberg :commit => '$GB_MOBILE_PR_REF'` => `gutenberg commit: '$GB_MOBILE_PR_REF'`

Reply:
I tried that and I think it's not compatible 😞 I think it only supports `:commit`

Reply:
Ah, good catch, thanks for testing it. Let's stick with this option then.

> We'll need to keep an eye on future release PRs since the tag: vs :commit change the : character position. Since before what I would do is just replace commit with tag 😅

Yep, this would be a bit confusing when updating the GB-mobile reference:
- When using commit SHA: `gutenberg :commit => '<SHA>'
- When using tag: `gutenberg tag: 'vX.Y.Z'